File: typical90/typical90_at.py
from collections import Counter


def main():
    Acnt = Counter(ai % D for ai in A)
    Bcnt = Counter(bi % D for bi in B)
    Ccnt = Counter(ci % D for ci in C)

    ans = 0
    for i in range(D):
        for j in range(D):
            k = (D-i-j) % D
            ans += Acnt[i] * Bcnt[j] * Ccnt[k]
    return print(ans)


if __name__ == '__main__':
    N, D = int(input()), 46
    A = list(map(int, input().split()))
    B = list(map(int, input().split()))
    C = list(map(int, input().split()))

    main()

File: typical90/typical90_at.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;
using vecll = vector<ll>;

#define rep(i, n) for (ll i = 0; i < ll(n); i++)

ll N;
const int D = 46;
vecll A, B, C;

int main() {
    cin >> N;
    A = vecll(N), B = vecll(N), C = vecll(N);
    rep(i, N) cin >> A[i];
    rep(i, N) cin >> B[i];
    rep(i, N) cin >> C[i];

    vecll Acnt(D, 0), Bcnt(D, 0), Ccnt(D, 0);
    rep(i, N) Acnt[A[i] % D]++, Bcnt[B[i] % D]++, Ccnt[C[i] % D]++;

    ll ans = 0;
    rep(i, D) rep(j, D) {
        ll k = (D * 2 - i - j) % D;
        ans += Acnt[i] * Bcnt[j] * Ccnt[k];
    }
    cout << ans << '\n';
    return 0;
}
